diagnostics/diag.go

```diff
@@ -336,3 +346,16 @@ func (d *Diagnostics) handleNewStream(s inet.Stream) {
 	d.HandleMessage(context.Background(), s)
 	s.Close()
 }
+
+func GetCommitHash() (string, error) {
+	gopath := os.Getenv("GOPATH")
+	ipfs := gopath + "/src/github.com/jbenet/go-ipfs"
```

Comment:
i wonder whether we could somehow make the build process go generate some code with the identifier. because they could be running a different binary-- if i switch branches with my daemon on, or pass a different GOPATH this doesnt work.